lib/puppet/file_bucket/dipper.rb

```diff
@@ -56,6 +58,38 @@ def backup(file)
     end
   end
 
+  # Diffs two filebucket files identified by their sums
+  def diff(suma, sumb)
+    raise RuntimeError, "Diff is not supported on windows platforms" if Puppet.features.microsoft_windows?
+    source_path = "#{@rest_path}#{@checksum_type}/#{suma}"
+    if sumb and sumb != ''
```

Comment:
Could we use `&&` here?  We try to avoid using `and` and `or` as logical operators.